src/us/kbase/workspace/test/database/provenance/CreditMetadataTest.java

```diff
@@ -0,0 +1,919 @@
+					.withLicense(nullOrWs)
+					.withVersion(nullOrWs)
+					.withDates(DATE_LIST)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, RESOURCE_TYPE,
+					CONTRIBUTOR_LIST, TITLE_LIST, EM,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+	}
+
+	@Test
+	public void buildMaximal() throws Exception {
+		for (Map.Entry<String, String> entry : VALID_PID_MAP.entrySet()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(entry.getKey(), RESOURCE_TYPE_STRING,
+							CONTRIBUTOR_LIST, TITLE_LIST)
+					.withLicense(LICENSE_STRING_UNTRIMMED)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.withComments(COMMENTS_LIST_UNTRIMMED)
+					.withDates(DATE_LIST)
+					.withFunding(FUNDING_LIST)
+					.withRelatedIdentifiers(RELATED_ID_LIST)
+					.build();
+			assertCreditMetadataFields(creditMetadata, entry.getValue(), RESOURCE_TYPE,
+					CONTRIBUTOR_LIST, TITLE_LIST, LICENSE_VERSION_MAP,
+					COMMENTS_LIST, DATE_LIST, FUNDING_LIST,
+					RELATED_ID_LIST);
+		}
+	}
+
+	@Test
+	public void buildMinimalOverwriteNullEmptyString() throws Exception {
+		for (final String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, RESOURCE_TYPE_STRING,
+							CONTRIBUTOR_LIST, TITLE_LIST)
+					.withLicense(LICENSE_STRING_UNTRIMMED)
+					.withLicense(nullOrWs)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.withVersion(nullOrWs)
+					.withDates(DATE_LIST)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, RESOURCE_TYPE,
+					CONTRIBUTOR_LIST, TITLE_LIST, EM,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+	}
+
+	@Test
+	public void buildMaximalOverwriteNullEmptyLists() throws Exception {
+		final CreditMetadata creditMetadata = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING, RESOURCE_TYPE_STRING,
+						CONTRIBUTOR_LIST, TITLE_LIST)
+				.withDates(DATE_LIST)
+				.withDates(Arrays.asList(null, null, null, null))
+				.withFunding(FUNDING_LIST)
+				.withFunding(null)
+				.withRelatedIdentifiers(RELATED_ID_LIST)
+				.withRelatedIdentifiers(ELRI)
+				.withVersion(VERSION_STRING_UNTRIMMED)
+				.build();
+		assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, RESOURCE_TYPE,
+				CONTRIBUTOR_LIST, TITLE_LIST, VERSION_MAP);
+	}
+
+	@Test
+	public void buildMaximalDedupeLists() throws Exception {
+		final CreditMetadata creditMetadata = CreditMetadata
+				.getBuilder(IDENTIFIER_STRING,
+						RESOURCE_TYPE_STRING,
+						CONTRIBUTOR_LIST_DUPES_NULLS,
+						TITLE_LIST_DUPES_NULLS)
+				.withComments(COMMENTS_LIST_UNTRIMMED)
+				.withDates(DATE_LIST_DUPES_NULLS)
+				.withFunding(FUNDING_LIST_DUPES_NULLS)
+				.withRelatedIdentifiers(RELATED_ID_LIST_DUPES_NULLS)
+				.build();
+				assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, RESOURCE_TYPE,
+				CONTRIBUTOR_LIST, TITLE_LIST, EM,
+				COMMENTS_LIST, DATE_LIST, FUNDING_LIST,
+				RELATED_ID_LIST);
+	}
```

Comment:
Did you want to test for immutability of the CM class? Should have caught this one earlier...

Reply:
Aren't all these classes going to need to be tested for immutability?

Reply:
I thought the ones that accepted or emitted mutable containers were tested? Did I miss some?

Either way, yeah if a value class accepts mutable containers or emits mutable containers testing that mutating the containers doesn't mutate the class should be done

Reply:
:+1: